x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/system_action_type_form.tsx

```diff
@@ -0,0 +1,340 @@
+
+  useEffect(() => {
+    (async () => {
+      if (disableErrorMessages) {
+        setActionParamsErrors({ errors: {} });
+        return;
+      }
+      const res: { errors: IErrorObject } = await actionTypeRegistry
+        .get(actionItem.actionTypeId)
+        ?.validateParams(actionItem.params);
+      setActionParamsErrors(res);
+    })();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [actionItem, disableErrorMessages]);
+
+  const actionTypeRegistered = actionTypeRegistry.get(actionConnector.actionTypeId);
+  if (!actionTypeRegistered) return null;
+
+  const showActionGroupErrorIcon = (): boolean => {
+    return !isOpen && some(actionParamsErrors.errors, (error) => !isEmpty(error));
+  };
+
+  const ParamsFieldsComponent = actionTypeRegistered.actionParamsFields;
+
+  const accordionContent = (
+    <>
+      <EuiSplitPanel.Inner color="plain">
+        {ParamsFieldsComponent ? (
+          <EuiErrorBoundary>
+            <EuiFlexGroup gutterSize="m" direction="column">
+              <EuiFlexItem>
+                <Suspense fallback={null}>
+                  <ParamsFieldsComponent
+                    actionParams={actionItem.params as any}
+                    errors={actionParamsErrors.errors}
+                    index={index}
+                    editAction={(key: string, value: RuleActionParam, i: number) => {
+                      setWarning(
+                        validateParamsForWarnings(
+                          value,
+                          http.basePath.publicBaseUrl,
+                          availableActionVariables
+                        )
+                      );
+                      setActionParamsProperty(key, value, i);
+                    }}
+                    messageVariables={aadTemplateFields}
+                    defaultMessage={defaultSummaryMessage}
+                    useDefaultMessage={true}
+                    actionConnector={actionConnector}
+                    executionMode={ActionConnectorMode.ActionForm}
+                    ruleTypeId={ruleTypeId}
+                  />
+                  {warning ? (
+                    <>
+                      <EuiSpacer size="s" />
+                      <EuiCallOut size="s" color="warning" title={warning} />
+                    </>
+                  ) : null}
+                </Suspense>
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </EuiErrorBoundary>
+        ) : null}
+      </EuiSplitPanel.Inner>
+    </>
+  );
+
+  return (
+    <>
+      <EuiSplitPanel.Outer hasShadow={isOpen}>
+        <EuiAccordion
+          initialIsOpen={true}
+          key={index}
+          id={index.toString()}
+          onToggle={setIsOpen}
+          paddingSize="none"
+          className="actAccordionActionForm"
+          buttonContentClassName="actAccordionActionForm__button"
+          data-test-subj={`alertActionAccordion-${index}`}
+          buttonContent={
```

Comment:
might be a good idea to move this buttonContent in a useMemo or even it is own component.